aten/src/ATen/CMakeLists.txt

```diff
@@ -16,6 +16,13 @@ IF(NOT AT_INSTALL_BIN_DIR OR NOT AT_INSTALL_LIB_DIR OR NOT AT_INSTALL_INCLUDE_DI
   SET(AT_INSTALL_SHARE_DIR "share" CACHE PATH "AT install include subdirectory")
 ENDIF()
 
+# this flag is used in Config but set externally, we must normalize it
+# to 0/1 otherwise `#if ON` will be evaluated to false.
+if (CAFFE2_STATIC_LINK_CUDA)
+  set(CAFFE2_STATIC_LINK_CUDA_INT 1)
+else()
+  set(CAFFE2_STATIC_LINK_CUDA_INT 0)
+endif()
```

Comment:
Ew. I'm surprised `#cmakedefine` doesn't handle this correctly

Reply:
Oh apparently we're not using `#cmakedefine`. Well, I guess we can fix this in follow up, if this CI round passes.

Reply:
We aren't using #cmakedefine for reasons described in the Config.h file.  This one took a long time to find. We can consider using #cmakedefine instead but it has the problem that forgetting to include Config.h is bad.

Reply:
i,i `#cmakedefine01` (but I think you told me fbcode makes this hard to do)